frontend/lib/src/features/password_reset/password_reset_page.dart

```diff
@@ -16,69 +23,137 @@ class PasswordResetPage extends StatelessWidget {
   const PasswordResetPage({super.key});
   @override
   Widget build(BuildContext context) {
+    //LoginController is being used here
+    //so that the button will disable and to avoid duplicated code
+    final LoginController loginController = Get.put(LoginController());
     final formKey = GlobalKey<FormBuilderState>();
+    final networkConfig = NetworkConfig();
+
+    void onSubmit() async {
+      // disable the button to prevent multiple requests being sent
+      loginController.setLoginButtonEnabled = false;
+
+      final client = networkConfig.client;
+
+      final email = formKey.currentState!.fields['email']!.value;
+
+      final forgotPasswordResponse = await client.post(
+        forgotPasswordUrl,
+        data: {
+          'email': email,
+        },
+      );
+
+      if (forgotPasswordResponse.statusCode != HttpStatus.badRequest) {
```

Comment:
```
if (forgotPasswordResponse.statusCode == HttpStatus.ok) {
        showAlertDialog(
          title: 'SUCCESS',
          content: 'The link was sent, please check your email',
          onClick: () {
            Get.to(ResetPasswordPage(emailValue: email));
          },
        );
      } else if (forgotPasswordResponse.statusCode == HttpStatus.badRequest) {
        // the error response is in Response<dynamic>, toString + jsonDecode to easily access data
        final error = jsonDecode(forgotPasswordResponse.data.toString());
        final message = error['error']['message']['email'][0];
        if (message == 'The selected email is invalid.') {
          formKey.currentState?.fields['email']?.invalidate(message);
        } else {
          showAlertDialog(
            title: 'ERROR',
            content: message,
            onClick: () {
              Get.to(ResetPasswordPage(emailValue: email));
            },
          );
        }
      }
```
      
      
      i just refactored your code a bit,

Reply:
my other comment did not send

currently if the error message is something other than the email validation

it will go to the reset password page

im just questioning why does it redirect to the next page if there is an error with the forgot password page?